File: readability.py
# Program that takes user to input of text and outputs a corresponding reading level

from cs50 import get_string

text = get_string("Text: ")
# Count number of letters in text ( all are upper + lowercase)


def count_letters(text):
    letters = 0
    # Starts at 1 since we will count the spaces to rep number of words
    words = 1
    sentences = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
        elif text[i] == " ":
            words += 1
        elif text[i] in [".", "!", "?"]:
            sentences += 1

    # Calculate average number of letters per 100 words store as L
    L = letters / words * 100
    S = sentences / words * 100
    # Input L into Coleman-Liau index equation (0.0588 * L - 0.296 * S - 15.8) to compute reading level
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Print reading level ("Grade: X"). If X < 1 print: "Before Grade 1". If x > 16 print: "Grade 16+"
    # Grade level below first
    if index < 1:
        print("Before Grade 1")
    # Grade level above 16
    elif index > 16:
        print("Grade 16+")
    # Grade levels between 1 & 16
    else:
        print("Grade ", index)


count_letters(text)

File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

//prototype of letter count
int count_letters(string text);

int main(void)
{
    // User input text
    string text = get_string("Text: ");
    //Display copy of text
    printf("%s\n", text);

    int count_letters(string text);
    int letters = 0;
    // words begin at a 1 count because for every 1 space there are 2 words (logical)
    int words = 1;
    int sentences = 0;
    //counting letters, words, & sentences
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        //counting spaces to count the words
        else if (text[i] == ' ')
        {
            words++;
        }
        //counting period, exclamation point, or question mark to indicate amount of sentences
        else if(text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }

    //calculating grade level
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

    //equation for grade level rounded
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    //grade level below first
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    //grade level above 16
    else if(index > 16)
    {
        printf("Grade 16+\n");
    }
    //grade levels between 1 & 16
    else
    {
        printf("Grade %i\n", index);
    }
}